pkg/execution/ratelimit/ratelimit.go

```diff
@@ -0,0 +1,66 @@
+package ratelimit
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/google/uuid"
+	"github.com/inngest/inngest/pkg/expressions"
+	"github.com/inngest/inngest/pkg/inngest"
+	"github.com/throttled/throttled/v2"
+	"github.com/xhit/go-str2duration/v2"
+)
+
+var (
+	ErrRateLimitExceeded = fmt.Errorf("rate limit exceeded")
+)
+
+type RateLimiter interface {
+	RateLimit(ctx context.Context, key string, c inngest.RateLimit) (bool, error)
+}
+
+// RateLimitKey returns the rate limiting key given a function ID, rate limit config,
+// and incoming event data.
+func RateLimitKey(ctx context.Context, id uuid.UUID, c inngest.RateLimit, evt map[string]any) (string, error) {
```

Comment:
should this be part of the key generator?

Reply:
The key generator is for the state store which is a separate package.  Rate limiting lives independently of that package.